examples/groth16-verifier/groth16/src/circom.rs

```diff
@@ -0,0 +1,269 @@
+// Copyright 2023 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::{anyhow, Error};
+use ark_bn254::{Bn254, Fr};
+use ark_groth16::{prepare_verifying_key, PreparedVerifyingKey, VerifyingKey};
+use serde::{Deserialize, Serialize};
+
+use crate::{convert_g1, convert_g2, from_u256};
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct CircomProof {
+    pub pi_a: Vec<String>,
+    pub pi_b: Vec<Vec<String>>,
+    pub pi_c: Vec<String>,
+    pub protocol: String,
+    pub curve: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct CircomVKey {
+    pub protocol: String,
+    pub curve: String,
+    #[serde(rename = "nPublic")]
+    pub n_public: u32,
+    pub vk_alpha_1: Vec<String>,
+    pub vk_beta_2: Vec<Vec<String>>,
+    pub vk_gamma_2: Vec<Vec<String>>,
+    pub vk_delta_2: Vec<Vec<String>>,
+    pub vk_alphabeta_12: Vec<Vec<Vec<String>>>,
+    #[serde(rename = "IC")]
+    pub ic: Vec<Vec<String>>,
+}
+
+impl CircomVKey {
+    pub fn pvk(&self) -> Result<PreparedVerifyingKey<Bn254>, Error> {
```

Comment:
```suggestion
    /// Prepare the verifier key for use by parsing the key and pre-computing values needed during verification.
    pub fn prepare_verifying_key(&self) -> Result<PreparedVerifyingKey<Bn254>, Error> {
```
Let's try to avoid using acronyms where we can since this code is meant to be read by users who want to know how it works.